plugins/inputs/docker/docker.go

```diff
@@ -366,9 +366,22 @@ func (d *Docker) gatherContainer(
 	var v *types.StatsJSON
 	// Parse container name
 	cname := "unknown"
-	if len(container.Names) > 0 {
-		// Not sure what to do with other names, just take the first.
-		cname = strings.TrimPrefix(container.Names[0], "/")
+	match := false
+	if len(container.Names) == 0 { // for tests
+		match = true
+	}
+
+	for i := range container.Names {
+		if !match {
+			match = d.containerFilter.Match(strings.TrimPrefix(container.Names[i], "/"))
+			if match {
+				cname = strings.TrimPrefix(container.Names[i], "/")
+			}
+		}
+	}
```

Comment:
I hate to be this specific, but this loop can be simplified:

```go
	var cname string
	for _, name := range container.Names {
		trimmedName = strings.TrimPrefix(name, "/")
		match := d.containerFilter.Match(trimmedName)
		if match {
			cname = trimmedName
			break
		}
	}

	if cname == "" {
		return nil
	}
```

By the way, its a minor thing but I never use the index form of the range loop unless I need it, indexing is harder to read, harder to type, and more prone to panic.